src/devices/came_top432.c

```diff
@@ -0,0 +1,169 @@
+    
+    if (decoder->verbose > 1) {
+        bitbuffer_printf(bitbuffer, "%s: ", __func__);
+    }
+
+    for (r = 0; r < bitbuffer->num_rows; ++r) {
+        b = bitbuffer->bb[r];
+
+        /*
+         * Validate message and reject invalid messages as
+         * early as possible before attempting to parse data.
+         *
+         * Check "message envelope"
+         * - valid message length (use a minimum length to account
+         *   for stray bits appended or prepended by the demod)
+         * - valid preamble/device type/fixed bits if any
+         * - Data integrity checks (CRC/Checksum/Parity)
+         */
+
+        if (bitbuffer->bits_per_row[r] < CAMETOP432_BITLEN) {
+            if (decoder->verbose > 1) {
+                fprintf(stderr, "%s: bitbuffer len received %d, expected : %d\n", __func__, bitbuffer->bits_per_row[r], CAMETOP432_BITLEN);
+            }
+            return DECODE_ABORT_EARLY;
+        }
+        // no preamble
+        // no crc
+        // no checksum
+        // no parity
+    }
+
+    // packet is not repeated
+    b = bitbuffer->bb[0];
+
+    // reconstruct the 12 bits code
+    code = ((b[0] << 4) | (b[1]>>4)) & 0xFFF;
+    sprintf(code_str, "%03x", code);
+    /* clang-format off */
+    data = data_make(
+            "brand",    "", DATA_STRING,   "CAME",
+            "model",    "", DATA_STRING,   "TOP432",
+            "type",     "", DATA_STRING,   "remote control",
+            "code",     "", DATA_INT,      code,
+            "code_hex", "", DATA_STRING,   code_str,
+            NULL);
+    /* clang-format on */
+    decoder_output_data(decoder, data);
+
+    // Return 1 if message successfully decoded
+    return 1;
+}
+
+/*
+ * List of fields that may appear in the output
+ *
+ * Used to determine what fields will be output in what
+ * order for this device when using -F csv.
+ *
+ */
+static char *output_fields[] = {
+        "brand",
+        "model",
+        "type",
+        "code",
+        "code_hex",
+        NULL,
+};
+
+/*
+ * r_device - registers device/callback. see rtl_433_devices.h
+ *
+ * Timings:
+ *
+ * short, long, and reset - specify pulse/period timings in [us].
+ *     These timings will determine if the received pulses
+ *     match, so your callback will fire after demodulation.
+ *
+ * Modulation:
+ *
+ * The function used to turn the received signal into bits.
+ * See:
+ * - pulse_demod.h for descriptions
+ * - r_device.h for the list of defined names
+ *
+ * This device is disabled and hidden, it can not be enabled.
+ *
+ * To enable your device, add it to the list in include/rtl_433_devices.h
+ * and to src/CMakeLists.txt and src/Makefile.am or run ./maintainer_update.py
+ *
+ */
+r_device came_top432 = {
+        .name        = "Came TOP432 remote control",
+        .modulation  = OOK_PULSE_PPM,
+        .short_width = 320,  // 
+        .long_width  = 640,  // 
+        .sync_width  = 320,
+        //.gap_limit   = 0,  // dont know how to find this value
```

Comment:
The amount of time in microseconds before we should consider a packet a new 'row', so should be something like 0.8 * `gap_size`, in which the gap size is the amount of 'dead time' between 2 packets.

Reply:
sync must never match short or long.

Reply:
ok, I think I confused reset_limit and gap_limit

Reply:
I'm really confused about values for `sync_width, gap_limit` and `reset_limit`. Here is a graphic representation of the signal.
![came_analysis](https://user-images.githubusercontent.com/14594410/89885925-1926b580-dbcc-11ea-9580-244d04aee896.png)
- ok, `short` and `long` width are clear
- `sync` is the sync start bit, but it should be the length of the short pulse (when measuring with Audacity there is variation of +- 5% on each pulse length...)
- `gap_limit` : it depends of what a "packet" is. Is it all the bits of the message, or a single bit ? The only working value for this decoder is "0".
- `reset_limit` : should be time between two messages (they are repeated 4 times). I measure 15 000µs.